weldx/utility.py

```diff
@@ -562,6 +562,129 @@ def xr_interp_like(
     return result
 
 
+def _check_dtype(var_dtype, ref_dtype) -> bool:
+    """Check if dtype matches a reference dtype (or is subdtype).
+
+    Parameters
+    ----------
+    var_dtype:
+        A numpy-dtype to test against.
+    ref_dtype
+        Python type or string description
+
+    Returns
+    -------
+    bool
+        True if dtypes matches.
+
+    """
+    if var_dtype != np.dtype(ref_dtype):
+        if isinstance(ref_dtype, str):
+            if (
+                "timedelta64" in ref_dtype
+                or "datetime64" in ref_dtype
+                and np.issubdtype(var_dtype, np.dtype(ref_dtype))
+            ):
+                return True
+
+        if not (
+            np.issubdtype(var_dtype, np.dtype(ref_dtype)) and np.dtype(ref_dtype) == str
+        ):
+            return False
+
+    return True
+
+
+def xr_check_coords(dax: xr.DataArray, ref: dict) -> bool:
+    """Validate the coordinates of the DataArray against a reference dictionary.
+
+    The reference dictionary should have the dimensions as keys and those contain
+    dictionaries with the following keywords (all optional):
+
+    ``values``
+        Specify exact coordinate values to match.
+
+    ``dtype`` : str or type
+        Ensure coordinate dtype matches at least one of the given dtypes.
+
+    ``optional`` : boolean
+        default ``False`` - if ``True``, the dimension has to be in the DataArray dax
+
+    Parameters
+    ----------
+    dax:
+        xarray object which should be validated
+    ref:
+        reference dictionary
+
+    Returns
+    -------
+    bool
+        True, if the test was a success, else throws an exception
```

Comment:
```suggestion
        True, if the test was a success, else an exception is raised
```

Nitpick: In python exceptions are raised :p